File: tests/e2e/runtime/array_overrides/java/postgresql/src/main/java/com/example/db/queries/Querier.java
package com.example.db.queries;

import com.example.db.models.Record;
import java.sql.Connection;
import java.sql.SQLException;
import java.time.LocalDateTime;
import java.util.Optional;
import java.util.UUID;
import javax.sql.DataSource;

public final class Querier {
    private final DataSource dataSource;

    public Querier(DataSource dataSource) {
        this.dataSource = dataSource;
    }

    public void insertRecord(String label, java.util.List<LocalDateTime> timestamps, java.util.List<UUID> uuids) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            Queries.insertRecord(conn, label, timestamps, uuids);
        }
    }

    public Optional<Record> getRecord(long id) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.getRecord(conn, id);
        }
    }
}

File: tests/e2e/runtime/array_overrides/kotlin/postgresql/src/main/kotlin/com/example/db/queries/Querier.kt
package com.example.db.queries

import com.example.db.models.Record
import java.time.LocalDateTime
import java.util.UUID
import javax.sql.DataSource

class Querier(private val dataSource: DataSource) {

    fun insertRecord(label: String, timestamps: List<LocalDateTime>, uuids: List<UUID>): Unit =
        dataSource.connection.use { conn -> Queries.insertRecord(conn, label, timestamps, uuids) }

    fun getRecord(id: Long): Record? =
        dataSource.connection.use { conn -> Queries.getRecord(conn, id) }
}
